Marlin/Marlin_main.cpp

```diff
@@ -5906,27 +5906,40 @@ inline void gcode_M410() { stepper.quick_stop(); }
 
   /**
    * M421: Set a single Mesh Bed Leveling Z coordinate
+   * Eter with 'M421 Xmm Ymm Zmm' or with 'M421 In Jm Zmm'
    */
   inline void gcode_M421() {
     float x = 0, y = 0, z = 0;
-    bool err = false, hasX, hasY, hasZ;
+    int8_t i = 0, j = 0;
+    bool err = false, hasX, hasY, hasZ, hasI, hasJ;
     if ((hasX = code_seen('X'))) x = code_value();
     if ((hasY = code_seen('Y'))) y = code_value();
+    if ((hasI = code_seen('I'))) i = code_value();
+    if ((hasJ = code_seen('J'))) j = code_value();
     if ((hasZ = code_seen('Z'))) z = code_value();
 
     if (hasX && hasY && hasZ) {
 
       int8_t ix = mbl.select_x_index(x),
              iy = mbl.select_y_index(y);
 
-      if (ix >= 0 && iy >= 0)
+      if (ix >= 0 && ix < MESH_NUM_X_POINTS && iy >= 0 && iy < MESH_NUM_Y_POINTS)
```

Comment:
`select_x_index` and `select_y_index` can never return a value >= to `MESH_NUM_X_POINTS` or `MESH_NUM_Y_POINTS`. If a point is too far out of range, these functions return a `-1`.